src/Language/Fixpoint/Solver/PLE.hs

```diff
@@ -1154,42 +1161,36 @@ instance Normalizable Equation where
 normalizeBody :: Symbol -> Expr -> Expr
 normalizeBody f = go
   where
-    go e
-      | f `elem` syms e
-      = go' e
-    go e
-      = e
-
-    go' (PAnd [PImp c e1,PImp (PNot c') e2])
-      | c == c' = EIte c e1 (go' e2)
-    go' e = e
-
-_splitBranches :: Symbol -> Expr -> [(Expr, Expr)]
-_splitBranches f = go
-  where
-    go (PAnd es)
-      | f `elem` syms es
-      = go' <$> es
-    go e
-      = [(PTrue, e)]
+    -- Q: Why do we only simplify the expression if f is a free symbol in it?
+    go e | f `elem` syms e = go' e
+    go e                   = e
 
-    go' (PImp c e) = (c, e)
-    go' e          = (PTrue, e)
+    -- | Simplifies a particular logical pattern:
+    -- (c => e1) /\ ((not c) => e2) --> if c then e1 else e2
+    --
+    -- Q: Why do we recurse into the right expression (only)?
```

Comment:
Question 2: Why do we recurse into the right expression (only)?

Reply:
Again, the expression has a concrete format (that came from Haskell's reflect annotations). The left expression is a conjuction of data constructor checkers, so you do not need to recurse there.

Reply:
again, may be good to add @nikivazou 's answer into the comment?

Reply:
This has been added in eaff89e a well.